student/src/main/java/tds/student/services/remote/RemoteOpportunityService.java

```diff
@@ -380,4 +402,70 @@ private static TestConfig mapExamConfigurationToTestConfig(ExamConfiguration exa
 
     return opportunitySegments;
   }
+
+  private List<TestSelection> mapAssessmentMetadataToTestSelections(final List<ExamAssessmentMetadata> assessmenMetadata,
+                                                                    final BrowserInfo browserInfo) throws ReturnStatusException {
+    List<TestSelection> selections = new ArrayList<>();
+
+    for (ExamAssessmentMetadata metadata : assessmenMetadata) {
+      TestSelection selection = new TestSelection();
+      selection.SetReturnStatus(new ReturnStatus(metadata.getStatus(), metadata.getDeniedReason()));
+      selection.setTestKey(metadata.getAssessmentKey());
+      selection.setTestID(metadata.getAssessmentId());
+      selection.setOpportunity(metadata.getAttempt());
+      selection.setMode("online");
+      selection.setDisplayName(metadata.getAssessmentLabel());
+      selection.setMaxOpportunities(metadata.getMaxAttempts());
+      selection.setSortOrder(0); // not used by UI
+      selection.setSubject(metadata.getSubject());
+      selection.setGrade(metadata.getGrade());
+      //TODO: Determine if we need this
+//      selection.getRequirements ().lookup (getTdsSettings ().getClientName (), testSelection);
+
+      validateBrowserInfo(browserInfo, selection);
+
+      selections.add(selection);
+    }
+
+    return selections;
+  }
+
+  private void validateBrowserInfo(final BrowserInfo browserInfo, final TestSelection selection) throws ReturnStatusException {
+    if (browserInfo != null) {
+      if (selection.getTestStatus() != TestSelection.Status.Disabled && selection.getTestStatus() != TestSelection.Status.Hidden) {
+        BrowserValidation browserValidation = new BrowserValidation();
+
+        for (BrowserRule rule : configRepository.getBrowserTestRules(selection.getTestID())) {
```

Comment:
Not without updating their ConfigRepository code, or creating a new endpoint in our ConfigService. Could do that, but for pragmatic purposes, and also because I don't think this is called often in practice, I that might be a bit of a premature optimization at the moment.